submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/submitter/request/SubmissionRequestFinalizer.kt

```diff
@@ -31,16 +30,15 @@ class SubmissionRequestFinalizer(
     }
 
     private fun deleteRemainingFiles(current: ExtSubmission?, previous: ExtSubmission) {
-        fun deleteFile(index: Int, file: ExtFile) {
-            logger.info { "${previous.accNo} ${previous.owner} Deleting file $index, path='${file.filePath}'" }
-            storageService.deleteSubmissionFile(previous, file)
-        }
-
         val subFiles = subFilesSet(current)
         logger.info { "${previous.accNo} ${previous.owner} Started deleting remaining submission files" }
-        serializationService.fileSequence(previous)
-            .filter { subFiles.contains(it.filePath).not() || it.storageMode != current?.storageMode }
-            .forEachIndexed { index, file -> deleteFile(index, file) }
+        storageService.deleteSubmissionFiles(
```

Comment:
And finally here, we would just call

```
storageService.deleteSubmissionFiles(
    sub = previous, 
    filter = { subFiles.contains(it.filePath).not() || it.storageMode != current?.storageMode },
)
```

WDYT? I think it makes the code easier to follow 

(3/3)